### DISCUSSION TITLE
Serving multiple paths on different domains

### DISCUSSION THREAD

**Discussion by Conorrr** [vercel/next.js#16170] [Help]
Hi,

Does anyone have any ideas how best to achieve the following?

I want to host a single next.js application on multiple domains, but I only want to serve a sub-path on each domain.
E.G.
The path for an incoming request `https://somedomain.com/some/path` gets converted by the reverse proxy to `localhost:3000/somedomaincom/some/path` so that multiple different domains can be hosted by a single instance of next.js. I am fine with the reverse proxy side of things and I can get next.js serving the correct sites on the right domains.

The issue I am facing is with routing.  I basically need the inverse of [customisable base paths](https://nextjs.org/blog/next-9-5#customizable-base-path). Either I am able to get links that load pages dynamically (and href links are incorrect as the contain the domain part) or the href links are correct but pages are not navigated to dynamically.

Hope that makes sense

Thanks
Conor

**Comment by Janpot** [comment]
History state can't be pushed across different origins. The next.js router can therefore not work across origins.

**Reply by Conorrr** [reply]
Sorry I should have been clearer. I only need to navigate around each domain, no need for cross-domain navigation.

**Comment by nickerlan** [comment]
@Conorrr have you found a solution?

**Reply by Conorrr** [reply]
I was unable to find a solution it seems that there is a fundamental assumption built into the router.

To me it seems that it would be really useful to host multiple sites from a single application. E.G. I could host a single application that with thousands of customers each with distinct domains.
